### ISSUE TITLE
linter: react/hook-use-state false positives for intentionally _unused getter or setter names

### ISSUE THREAD

**Issue by justingrant** [oxc-project/oxc#24952]
### What version of Oxlint are you using?

1.73.0

### What command did you run?

`oxlint -c .oxclint.json`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["react"],
  "rules": {
    "react/hook-use-state": "error"
  }
}
```

### What happened?

A common pattern is to have write-only (or, less commonly, read-only) use of the `useState` hook.  This rule should allow common "this is intentionally unused" patterns like the underscore prefix. For example, I'd like the code below to not trigger this rule.

```ts
const Component2 = () => {
  const [_, id] = useState(1);
}
const Component1 = () => {
  const [id, _] = useState(1);
}
```

**Comment by baevm** [comment]
I think both examples with underscore also reported by the upstream `eslint-plugin-react/hook-use-state` and `@eslint-react/naming-convention/use-state` rules? There was a related discussion (but case with omitting setter entirely) in`@eslint-react` https://github.com/Rel1cx/eslint-react/issues/1342, i think it was fixed with allowing `const [id] = useState(1)`

**Comment by camc314** [comment]
lets make this a config option tht defaults to the same as the upstream behaviour.

From the examples you've provided I don't understand why you'd want this. Like why not just do:
```
const id = 1
const Component1 = () => {
  ...
}
```

**Comment by justingrant** [comment]
> From the examples you've provided I don't understand why you'd want this

Here's some existing code (not mine) in a project I was migrating from ESLint.

I'm not claiming this is good code (and these should almost certainly be rewritten to use idiomatic React patterns) but `react/hook-use-state` lint rule (focused on naming and destructuring) is IMO the wrong place to complain. Misuse of read-only/write-only `useState` seems like a good candidate for a new rule!

```ts
  const [_, setRenderCount] = useState(0);

  useEffect(() => {
    if (viewerState.cameraOrtho) {
      cameraRef.current = orthographicCameraRef.current;
    } else {
      cameraRef.current = perspectiveCameraRef.current;
    }
    // When the projection mode changes, a new camera is used and the camera
    // ref passed to the OrbitControls must be updated. By incrementing the
    // renderCount, a re-render is triggered using the updated cameraRef.
    setRenderCount((n) => n + 1);
  }, [viewerState.cameraOrtho]);

  // more code below elided
```

```ts
export const Default: StoryFn<typeof CommentBox> = (_args) => {
  const [comment, _] = useState(mockComment(lorem));

  return <CommentBox comment={comment} />;
};
```

```ts
  // When generating a random ID if none is provided, cache the value to
  // prevent unnecessary re-renders.
  const [id, _] = useState(restProps.id || crypto.randomUUID());
```
